File: src/main/java/model/Anunciante.java
package model;

import kontroller.Usuario;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Entity
public class Anunciante implements Serializable {
	private static final long serialVersionUID = 3943662055685412967L;

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO, generator = "native")
	@GenericGenerator(name = "native", strategy = "native")
	private Long id;
	private String nome;
	@Transient
	private String cidade;
	@OneToOne(fetch = FetchType.LAZY, cascade = CascadeType.PERSIST)
	private Usuario usuario;
	@OneToMany(fetch = FetchType.EAGER)
	private List<Vaga> vagas;


	public Anunciante() {
	}

	public Anunciante(String nome, Usuario usuario) {
		this.nome = nome;
		this.usuario = usuario;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public List<Vaga> getVagas() {
		return vagas;
	}

	public void setVagas(List<Vaga> vagas) {
		this.vagas = vagas;
	}

	public Usuario getUsuario() {
		return usuario;
	}

	public void setUsuario(Usuario usuario) {
		this.usuario = usuario;
	}
}

File: src/main/java/model/Anunciante.kt
package model

import org.hibernate.annotations.GenericGenerator
import java.io.Serializable
import javax.persistence.*

@Entity
class Anunciante : Serializable {

    @Transient
    private val serialVersionUID = 5911361446123528715L


    @Id
    @GeneratedValue(strategy = GenerationType.AUTO, generator = "native")
    @GenericGenerator(name = "native", strategy = "native")
    var id: Long? = null
    var nome: String = ""
    @OneToOne(fetch = FetchType.LAZY, cascade = [CascadeType.PERSIST])
    var usuario: Usuario? = null
    @OneToMany(fetch = FetchType.EAGER)
    var vagas: List<Vaga>? = ArrayList()

    constructor()

    constructor(nome: String, usuario: Usuario) {
        this.nome = nome
        this.usuario = usuario
    }

}